processor/src/trace/virtual_bus/sum_check/verifier/mod.rs

```diff
@@ -0,0 +1,152 @@
+    final_query_builder: V,
+    _challenger: PhantomData<C>,
+}
+
+impl<B, E, P, C, H, V> SumCheckVerifier<B, E, P, C, H, V>
+where
+    B: StarkField,
+    E: FieldElement<BaseField = B>,
+    C: RandomCoin<Hasher = H, BaseField = B>,
+    P: CompositionPolynomial<E>,
+    H: ElementHasher<BaseField = B>,
+    V: FinalQueryBuilder<Field = E>,
+{
+    /// Create a new [SumCheckVerifier] from a composition polynomial and final query builder.
+    pub fn new(composition_poly: P, final_query_builder: V) -> Self {
+        let max_degree = composition_poly.max_degree();
+        let eval_domain = EvaluationDomain::new(max_degree);
+
+        Self {
+            composition_poly,
+            eval_domain,
+            final_query_builder,
+            _challenger: PhantomData,
+        }
+    }
+
+    /// Verifies a sum-check proof [Proof] and returns a claim on the openings of the mult-linear
+    /// oracles that are part of the statement being proven.
+    ///
+    /// More precisely, the method:
+    ///
+    /// 1. Generates a `claimed_evaluation` from the round proof polynomials and the round challenge
+    /// randomness.
+    ///
+    /// 2. Computes a query that is built using the [FinalQueryBuilder] from the multi-linear
+    /// openings and the round challenges.
+    ///
+    /// 3. Evaluates the composition polynomial at the query and checks that it is equal
+    /// `claimed_evaluation`.
+    ///
+    /// 4. Outputs a `FinalOpeningClaim` on the multi-linear oracles.
+    ///
+    /// Thus, the proof is correct if the method outputs a [FinalOpeningClaim] and this latter is
+    /// a valid claim on the multi-linear oracles i.e., each multi-linear oracle opens to the
+    /// claimed value at the specified opening point.
+    ///
+    ///  # Errors
+    /// Returns an error if:
+    /// - No openings were provided.
+    /// - Draw the round challenge fails.
+    /// - The final evaluation check fails.
+    pub fn verify(
+        &self,
+        claim: E,
+        proof: Proof<E>,
+        coin: &mut C,
+    ) -> Result<FinalOpeningClaim<E>, Error> {
+        let Proof {
+            openings: openings_claim,
+            round_proofs,
+        } = proof;
+
+        let mut claimed_evaluation = claim;
+        let mut evaluation_point = vec![];
+        for round_proof in round_proofs {
+            let partial_evals = round_proof.poly_evals.clone();
+            coin.reseed(H::hash_elements(&partial_evals));
+            let evals = round_proof.to_evals(claimed_evaluation);
+
+            let r = coin.draw().map_err(|_| Error::FailedToGenerateChallenge)?;
+            let reduced_evaluation = self.eval_domain.evaluate(&evals, r);
+
+            claimed_evaluation = reduced_evaluation;
+            evaluation_point.push(r);
+        }
+
+        if let Some(openings_claim) = openings_claim {
+            if openings_claim.evaluation_point != evaluation_point {
+                return Err(Error::WrongOpeningPoint);
+            }
```

Comment:
Is this check necessary? The prover can just put anything in `openings_claim.evaluation_point`, unrelated to the openings. So IIUC, we could just remove the field `OpeningClaims::evaluation_point`.

Reply:
You are right but there are two reasons for having it as part of the claim:

1. In the case where we have many different runs of different sum-checks, we would have different opening points against some of the same oracles. At this point, one can usually do a batch opening argument in order to reduce the cost of the opening proof. This is probably not immediately relevant for us but could be in the future.
2. The more relevant reason for us is so that the STARK prover would have all the data necessary in order to build the auxiliary trace helper columns in order to do the (multi-linear) opening proof.

To summarize, we could, and probably, should remove the evaluation points (in order to reduce the size of the GKR proofs), but having it this way is convenient at the moment and potentially necessary in the future.